src/Compilers/Core/Portable/SourceGeneration/Nodes/TransformNode.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class TransformNode<TInput, TOutput> : IIncrementalGeneratorNode<TOutput>
+    {
+        private readonly Func<TInput, IEnumerable<TOutput>> _func;
+        private readonly IIncrementalGeneratorNode<TInput> _sourceNode;
+
+        public TransformNode(IIncrementalGeneratorNode<TInput> sourceNode, Func<TInput, TOutput> userFunc)
+            : this(sourceNode, userFunc: (i) => ImmutableArray.Create(userFunc(i)))
+        {
+        }
+
+        public TransformNode(IIncrementalGeneratorNode<TInput> sourceNode, Func<TInput, IEnumerable<TOutput>> userFunc)
+        {
+            _sourceNode = sourceNode;
+            _func = userFunc;
+        }
+
+        // PROTOTYPE(source-generators):
```

Comment:
It gets passed to `ModifyEntriesFromPreviousTable` to be used instead of the default comparer as it is today.

---
In reply to: [619512145](https://github.com/dotnet/roslyn/pull/52809#discussion_r619512145) [](ancestors = 619512145)